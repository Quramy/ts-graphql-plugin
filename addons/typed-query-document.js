module.exports = require('../lib/typegen-addons').TypedQueryDocumentAddonFactory;
