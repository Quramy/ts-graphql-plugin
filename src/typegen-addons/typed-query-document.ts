import ts from 'typescript';
import { TypeGenAddonFactory } from '../typegen';

export const TypedQueryDocumentAddonFactory: TypeGenAddonFactory = ({ source }) => ({
  operationDefinition({ tsResultNode, tsVariableNode }) {
    const lhs = ts.createIdentifier(`${tsResultNode.name.text}Document`);
    const rhs = ts.createTypeReferenceNode(ts.createIdentifier('TypedQueryDocumentNode'), [
      ts.createTypeReferenceNode(ts.createIdentifier(tsResultNode.name.text)),
      ts.createTypeReferenceNode(ts.createIdentifier(tsVariableNode.name.text)),
    ]);
    const modifiers = [ts.createModifier(ts.SyntaxKind.ExportKeyword)];
    const typeAliasDeclaration = ts.createTypeAliasDeclaration(undefined, modifiers, lhs, undefined, rhs);
    source.pushNamedImportIfNeeded('TypedQueryDocumentNode', 'graphql');
    source.pushStatement(typeAliasDeclaration);
  },
});
