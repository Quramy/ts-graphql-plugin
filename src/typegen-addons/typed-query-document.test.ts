import { TypedQueryDocumentAddonFactory } from './typed-query-document';
import { createAddonTester } from './testing/addon-tester';

describe(TypedQueryDocumentAddonFactory, () => {
  const addonTester = createAddonTester(TypedQueryDocumentAddonFactory);

  const sdl = `
    type Query {
      hello: String!
    }
  `;

  it('should add export statement using TypedQueryDocumentNode', () => {
    const { outputSourceFiles } = addonTester.generateTypes({
      files: [{ fileName: 'main.ts', content: 'const query = `query MyQuery { hello }`' }],
      schemaSDL: sdl,
    });
    expect(outputSourceFiles.map(f => f.content)).toMatchSnapshot();
  });
});
