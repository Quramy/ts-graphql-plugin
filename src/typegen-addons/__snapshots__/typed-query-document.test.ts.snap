// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypedQueryDocumentAddonFactory should add export statement using TypedQueryDocumentNode 1`] = `
Array [
  "/* eslint-disable */
/* This is an autogenerated file. Do not edit this file directly! */
import { TypedQueryDocumentNode } from \\"graphql\\";
export type MyQuery = {
    hello: string;
};
export type MyQueryVariables = {};
export type MyQueryDocument = TypedQueryDocumentNode<MyQuery, MyQueryVariables>;
",
]
`;
